Source/Scene/CreditDisplay.js

```diff
@@ -376,10 +399,18 @@ CreditDisplay.prototype.addCredit = function (credit) {
     return;
   }
 
+  let credits;
   if (!credit.showOnScreen) {
-    this._currentFrameCredits.lightboxCredits.set(credit.id, credit);
+    credits = this._currentFrameCredits.lightboxCredits;
+  } else {
+    credits = this._currentFrameCredits.screenCredits;
+  }
+
+  if (credits.contains(credit.id)) {
+    const creditCount = credits.get(credit.id).count;
+    credits.set(credit.id, new CreditDisplayElement(credit, creditCount + 1));
   } else {
-    this._currentFrameCredits.screenCredits.set(credit.id, credit);
+    credits.set(credit.id, new CreditDisplayElement(credit));
   }
```

Comment:
I should have noticed this sooner. Both the tuple approach and struct approach will allocate a new object for each unique credit that's added. We should avoid that.

One fix is to have two separate `AssociateArray`, one for the credits and other for the counts, as discussed offline. Another is to recycle `CreditDisplayElement` objects using a `ManagedArray`.